cmd/thanos/compact.go

```diff
@@ -480,6 +490,12 @@ func (cc *compactConfig) registerFlag(cmd *kingpin.CmdClause) *compactConfig {
 		"as querying long time ranges without non-downsampled data is not efficient and useful e.g it is not possible to render all samples for a human eye anyway").
 		Default("false").BoolVar(&cc.disableDownsampling)
 
+	cmd.Flag("compaction.disable", "Disables compaction. This is not recommended "+
+		"as compaction optimizes the stored data in bucket").
+		Default("false").BoolVar(&cc.disableCompaction)
+
+	cmd.Flag("retention.disable", "Disables retention.").Default("false").BoolVar(&cc.disableRetention)
```

Comment:
we don't need this as default doesn't do any retention `0d` value

Reply:
Yeh, but there is still the process of deleting already marked blocks for deletion which happens regardless on the resolution retention.

Reply:
maybe then let's change the flag to disable just the marked blocks?

Reply:
I don't like having two ways doing the same thing.